test-data/unit/check-protocols.test

```diff
@@ -2890,6 +2890,12 @@ class SupportsStr(Protocol):
 def ss(s: SupportsStr) -> None: pass
 ss(None)
 
+class SupportsBool(Protocol):
+    def __bool__(self) -> bool: ...
```

Comment:
Please, add two more protocols to the test: `def __bool__(self) -> Literal[True]: ...` and `def __bool__(self) -> Literal[False]: ...`

Other than that - it looks good to me.

Reply:
Thanks for review! I add tests with `Literal` return types, take a look, please